src/memusage.h

```diff
@@ -177,23 +176,33 @@ static inline size_t DynamicUsage(const std::list<X>& l)
     return MallocUsage(sizeof(list_node<X>)) * l.size();
 }
 
+// Empirically, an std::unordered_map node has two pointers (likely
+// forward and backward pointers) on some platforms (Windows and macOS),
+// so be conservative in estimating memory usage by assuming this is
+// the case for all platforms.
 template<typename X>
 struct unordered_node : private X
 {
 private:
     void* ptr;
+    void* ptr2;
 };
 
+// The memory used by an unordered_set or unordered map is the sum of the
+// sizes of the individual nodes (which are separately allocated) plus
+// the size of the bucket array (which is a single allocation).
+// Empirically, each element of the bucket array consists of two pointers
+// on some platforms (Windows and macOS), so be conservative.
 template<typename X, typename Y>
 static inline size_t DynamicUsage(const std::unordered_set<X, Y>& s)
 {
-    return MallocUsage(sizeof(unordered_node<X>)) * s.size() + MallocUsage(sizeof(void*) * s.bucket_count());
+    return MallocUsage(sizeof(unordered_node<X>)) * s.size() + MallocUsage(2 * sizeof(void*) * s.bucket_count());
 }
 
 template<typename X, typename Y, typename Z>
 static inline size_t DynamicUsage(const std::unordered_map<X, Y, Z>& m)
 {
-    return MallocUsage(sizeof(unordered_node<std::pair<const X, Y> >)) * m.size() + MallocUsage(sizeof(void*) * m.bucket_count());
+    return MallocUsage(sizeof(unordered_node<std::pair<const X, Y> >)) * m.size() + MallocUsage(2 * sizeof(void*) * m.bucket_count());
```

Comment:
`2 * sizeof(void*)` refers to `ptr` + `ptr2`, right? Can we extract that to an explanatory constant?

Reply:
Yes, and there is a fairly detailed comment on lines 194-195 (just before the `unordered_set` version of `DynamicUsage`), so I think that's sufficient. But if you feel strongly, I'll try to come up with something.